runtime/oti/vmconstantpool.xml

```diff
@@ -80,7 +80,7 @@
 	<classref name="java/lang/ClassFormatError"/>
 	<classref name="java/lang/StackTraceElement"/>
 	<classref name="java/lang/StackWalker$StackFrameImpl"/>
-	
+	<classref name="java/lang/InternalRamClass"/>
```

Comment:
Although adding classes to the vmconstantpool is the easy way to get them loaded / available, it is a startup cost. 

Given that the sun_reflect_constantpool code already uses the JNI id cache, wouldn't this be a good candidate for that mechanism rather than the vmconstantpool?

Reply:
Do you mean load j.l.InternalRamClass with `(*env)->FindClass(...)` and then store it in the `JniIDCache` ?

If we took this approach we would have to also cache the `vmref` `jfieldID` as well and use `(*env)->GetLongField(...)` in every place we are using `J9CLASS_FROMJ9CLASSREF`. This would mean adding more acquire / releases vmaccess points.

What I really need is the vmconstantpool fieldref for `vmref`. It is my understanding that if you create a fieldref you also need the owning classref in the vmconstanpool. I don't need to explicitly load the class or even require `J9Class*` for j.l.InternalRamClass.

Reply:
Correct - anything in the VM CP with a fieldref is resolved on startup, and can always be used.  You still need VM access to do so.

Also note, that JNI functions like FindClass are VM access acquire/release points already, so doing it explicitly is not actually more calls.